code/ndarray.c

```diff
@@ -281,13 +282,13 @@ size_t true_length(mp_obj_t bool_list) {
     return trues;
 }
 
-mp_bound_slice_t generate_slice(mp_uint_t n, mp_obj_t index) {
+static mp_bound_slice_t generate_slice(mp_uint_t n, mp_obj_t index) {
     // micropython seems to have difficulties with negative steps
     mp_bound_slice_t slice;
     if(MP_OBJ_IS_TYPE(index, &mp_type_slice)) {
         mp_obj_slice_indices(index, n, &slice);
     } else if(MP_OBJ_IS_INT(index)) {
-        int32_t _index = mp_obj_get_int(index);
+        size_t _index = mp_obj_get_int(index);
```

Comment:
I think I should have commented here. int32_t is probably necessary, since _index < 0 is possible.

Reply:
Fixed in 86e7155ab7b72ce3cc63d29b4b053f5095061412